tdp_core/settings/model.py

```diff
@@ -111,14 +111,14 @@ class GlobalSettings(BaseSettings):
     def is_development_mode(self) -> bool:
         return self.env.startswith("dev")
 
-    def get_nested(self, key: str, default: Any = None) -> Union[Any, None]:
+    def get_nested(self, key: str, default: Optional[Dict] = None) -> Dict:
```

Comment:
How come the default value should be `Dict`? What if I want the leaf value which could be anything from `str` to `int` to `bool` etc.?